actors/runtime/src/main/java/cloud/orbit/actors/runtime/Hosting.java

```diff
@@ -392,7 +392,8 @@ private void updateServerNodes()
         final DistributedMap<RemoteKey, NodeAddress> distributedDirectory = getDistributedDirectory();
 
         // Get the existing activation from the distributed cache (if any)
-        NodeAddress nodeAddress = await(distributedDirectory.get(remoteKey));
+        NodeAddress nodeAddress = await(distributedDirectory.get(remoteKey)
+                .whenCompleteAsync((r, e) -> {}, stage.getExecutionPool()));
```

Comment:
In some cases the distributedDirectory implementation may return on its own callback thread and may be blocked in selectNode waiting for a node to join the cluster hosting a particular actor.